ZenPacks/zenoss/Microsoft/Windows/datasources/ServiceDataSource.py

```diff
@@ -208,50 +201,76 @@ def collect(self, config):
 
         defer.returnValue(results)
 
+    def buildServicesDict(self, datasources):
+        services = {}
+        for ds in datasources:
+            services[ds.params['servicename']] = {'eventClass': ds.eventClass,
+                                                  'eventKey': ds.eventKey,
+                                                  'severity': ds.severity,
+                                                  'alertifnot': ds.params['alertifnot']}
+        return services
+
     def onSuccess(self, results, config):
 
         data = self.new_data()
-        ds0 = config.datasources[0]
-        serviceinfo = results[results.keys()[0]]
-        eventClass = ds0.eventClass if ds0.eventClass else "/Status"
-        eventKey = ds0.eventKey if ds0.eventKey else "WindowsService"
-
-        if serviceinfo[0].State != ds0.params['alertifnot']:
-
-            evtmsg = 'Service Alert: {0} has changed to {1} state'.format(
-                serviceinfo[0].Name,
-                serviceinfo[0].State
-            )
-
+        services = self.buildServicesDict(config.datasources)
+        log.debug('Windows services query results: {}'.format(results))
+        try:
+            serviceinfo = results[results.keys()[0]]
+        except:
             data['events'].append({
-                'eventClass': eventClass,
-                'eventClassKey': 'WindowsServiceLog',
-                'eventKey': eventKey,
-                'severity': ds0.severity,
-                'summary': evtmsg,
-                'component': prepId(serviceinfo[0].Name),
-                'device': config.id,
-            })
-        else:
-
-            evtmsg = 'Service Recoverd: {0} has changed to {1} state'.format(
-                serviceinfo[0].Name,
-                serviceinfo[0].State
-            )
-
-            data['events'].append({
-                'eventClass': eventClass,
-                'eventClassKey': 'WindowsServiceLog',
-                'eventKey': eventKey,
-                'severity': ZenEventClasses.Clear,
-                'summary': evtmsg,
-                'component': prepId(serviceinfo[0].Name),
-                'device': config.id,
-            })
+                                'eventClass': "/Status",
+                                'severity': ZenEventClasses.Warning,
```

Comment:
This seems more like an Error severity than Warning. As would the same down in the onError method.